juntagrico/views.py

```diff
@@ -355,22 +355,24 @@ def profile(request):
 def cancel_membership(request):
     member = request.user.member
     if request.method == 'POST':
-        now = timezone.now().date()
-        end_date = next_membership_end_date()
-        message = request.POST.get('message')
-        member = request.user.member
-        member.end_date = end_date
-        member.cancellation_date = now
-        if member.is_cooperation_member:
-            adminnotification.member_canceled(member, end_date, message)
-        else:
-            member.deactivation_date = now
-        member.save()
-        for share in member.active_shares:
-            cancel_share(share, now, end_date)
-        return redirect('profile')
-
-    missing_iban = member.iban == ''
+        form = MemberCancellationForm(member, request.POST)
+        if form.is_valid():
+            now = timezone.now().date()
+            end_date = next_membership_end_date()
+            message = form.cleaned_data['message']
+            member = request.user.member
+            member.end_date = end_date
+            member.cancellation_date = now
+            if member.is_cooperation_member:
+                member.iban = form.cleaned_data['iban']
+                adminnotification.member_canceled(member, end_date, message)
+            else:
+                member.deactivation_date = now
```

Comment:
This will fail if the member has a cancelled but not yet deactivated subscription.

Reply:
taken into consideration in new code